rotkehlchen/api/v1/schemas.py

```diff
@@ -2115,6 +2116,25 @@ class AssetsReplaceSchema(Schema):
     source_identifier = fields.String(required=True)
     target_asset = AssetField(required=True, expected_type=Asset, form_with_incomplete_data=True)
 
+    @validates_schema
+    def validate_schema(
+            self,
+            data: dict[str, Any],
+            **_kwargs: Any,
+    ) -> None:
+        try:
+            source_is_evm_token = Asset(data['source_identifier']).is_evm_token()
+        except UnknownAsset:
+            can_merge = True  # we allow to merge assets that are not in the globaldb
+        else:
+            can_merge = not (source_is_evm_token and data['target_asset'].is_evm_token())
```

Comment:
This can fail. If we get to the except above, `source_is_evm_token` will not exist. Weird that linters did not see this

Reply:
The else executes when there is no exception raised. Finally is what always executes

Reply:
ah yes indeed. But read the other comment :P